Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls.h

```diff
@@ -594,16 +594,45 @@ static bool HasSyscallError(const void* Result) {
 template<bool IncrementOffset, typename T>
 uint64_t GetDentsEmulation(int fd, T* dirp, uint32_t count);
 
-namespace FaultSafeMemcpy {
+namespace FaultSafeUserMemAccess {
   // These are little helper functions for cases when FEX needs to copy data to or from the application in a robust fashion.
   // CopyFromUser and CopyToUser are memcpy routines that expect to safely SIGSEGV when reading or writing application memory respectively.
   // Returns zero if the memcpy completed, or crashes with SIGABRT and a log message if it faults.
   [[nodiscard]]
   size_t CopyFromUser(void* Dest, const void* Src, size_t Size);
   [[nodiscard]]
   size_t CopyToUser(void* Dest, const void* Src, size_t Size);
+#if defined(ASSERTIONS_ENABLED) && ASSERTIONS_ENABLED && defined(_M_ARM_64)
+  // These helpers just check if the user pointer is readable and writable.
+  // This is useful in an assert build that can be safely sprinkled through the syscall handler without overhead in release builds.
+  void VerifyIsReadable(const void* Src, size_t Size);
+  void VerifyIsReadableOrNull(const void* Src, size_t Size);
+  void VerifyIsWritable(void* Src, size_t Size);
+  void VerifyIsWritableOrNull(void* Src, size_t Size);
+
+  // Iterates a string and checks if all bytes are readable. Stops at nullptr.
+  void VerifyIsStringReadable(const char* Src);
+
+  // Iterates a string and checks if all bytes are readable. Stops at nullptr or the maxsize provided.
+  void VerifyIsStringReadableMaxSize(const char* Src, size_t MaxSize);
+#else
+  inline void VerifyIsReadable(const void* Src, size_t Size) {
+    LOGMAN_THROW_A_FMT(Src != nullptr, "Not null");
```

Comment:
`LOGMAN_THROW_A_FMT` is a no-op in this case, so another assertion method must be used.

Reply:
Switched over to `ERROR_AND_DIE_FMT`